.github/workflows/nix-ci.yml

```diff
@@ -0,0 +1,61 @@
+name: llvm-pretty-bc-parser nix CI
+on:
+  push:
+    branches:
+      - 'master'
+      - 'release-**'
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        ghc-version: [ "ghc94", "ghc92", "ghc810", "ghc88" ]
```

Comment:
It's interesting that this has an entirely separate build matrix from `build.yml`. My understanding is that this is primarily meant to test different LLVM versions against `disasm-test`, so is there much of a point to trying this step with different GHC versions?

Reply:
It wasn't hard to add this and seemed like a good backup to the other build channel, even though it's somewhat of a duplication.  This is also a replication of a similar CI yaml file I've used elsewhere where those locations don't have a parallel path, so this keeps the diff cleaner between those and this.

Reply:
> even though it's somewhat of a duplication

Indeed, `nix-ci.yml` does nearly everything that `build.yml` does, plus additional things. I'm wondering if we shouldn't just remove `build.yml` in favor of `nix-ci.yml`.

Reply:
It's a possibility.  I'd suggest deferring that decision a little until I get some of these regression tests added and then make this determination.  I am also probably not the best qualified to vote on this, being very comfortable with the nix approach which is not necessarily the case for others, and the `build.yml` is more conventional.

Reply:
Fair enough.